raiden/storage/sqlite.py

```diff
@@ -363,6 +363,16 @@ def count_state_changes(self) -> int:
 
         return int(result[0][0])
 
+    def count_snapshots(self) -> int:
+        cursor = self.conn.cursor()
+        query = cursor.execute("SELECT COUNT(1) FROM state_snapshot")
+        result = query.fetchall()
+
+        if len(result) == 0:
+            return 0
+
+        return int(result[0][0])
```

Comment:
Since this is only used to check if there are any snapshots at all, you could use `SELECT EXISTS(SELECT 1 FROM state_snapshot)`. which won't have to read all rows in the table.